### DISCUSSION TITLE
Implementing ASCPEM

### DISCUSSION THREAD

**Discussion by ArneTR** [green-code-initiative/ecoCode#193] [Ideas]
Hey ecoCode Team,

I have seen the page [docs page](https://github.com/green-code-initiative/ecoCode/tree/main/docs/rules) about the rules that are planned to be implemented.

I was curios if you also know about ASCPEM and if you plan to implement that or not. And if so, why.
Especially I would be interested in the overlap or reduancy of these rules.

To my understanding ASCPEM is the closest one to a standard. Some companies like NTT DATA [even claim to have it already implemented](https://greensoftware.foundation/articles/how-intesa-and-ntt-data-measure-energy-consumption-of-software#_msocom_1). Have you maybe reached out to them to see if something of that work is open source and can be reused?

**Comment by dedece35** [comment]
Hi @ArneTR,

thank you for idea. Sorry, I don't know ASCPEM but why not upgrade our rules referential.
What do you think about it @glalloue, @mdubois81, @olegoaer ?
We will discuss about your idea during our internal meeting and give you a feedback.

**Comment by olegoaer** [comment]
Hello,
Ironically, having studied the OMG standards, I completely missed this specification. Thanks for the reference @ArneTR (we already handshaked by email a few month ago ;)

I'll take a look. As always, the point is to know what comes under performance (i.e. speed) and what comes under carbon footprint. We also need to look at what is statically detectable and what is not.

Olivier Le Goaër

**Comment by olegoaer** [comment]
hi @ArneTR 

Further to our discussion, it turns out that the OMG has recently been working on a specification much closer to what we want to do with ecoCode: [ASCRSM](https://www.omg.org/spec/ASCRSM/1.0/Beta1/About-ASCRSM).

**Comment by ArneTR** [comment]
Intereting to see that a new of these specs has been released.

Do you have any insight why the OMG is creating these standards in the first place? I have so far never seen any payment (like for ISO standards) or any organization that is implementing them.

On a different note: The standard looks very similar to ASCPEM. I have so far not seen any proof why these patterns are specifically energy intensive. Is there any data to back this up?

What I would be super interested is also how EcoCode does this. How do you validate if the rules that are checked actually have a gain or even a net gain (meaning that cost of finding the deficiency is not higher than the reduction)?
